code/datums/components/acid.dm

```diff
@@ -190,24 +202,25 @@ GLOBAL_DATUM_INIT(acid_overlay, /mutable_appearance, mutable_appearance('icons/e
 	return NONE
 
 /// Handles searing the hand of anyone who tries to touch this without protection.
-/datum/component/acid/proc/on_attack_hand(atom/parent_atom, mob/living/carbon/user)
+/datum/component/acid/proc/on_attack_hand(atom/source, mob/living/carbon/user)
 	SIGNAL_HANDLER
 
 	if(!istype(user))
 		return NONE
-	if((parent_atom == user) || (parent_atom.loc == user))
+	if((source == user) || (source.loc == user))
 		return NONE // So people can take their own clothes off.
 	if((acid_power * acid_volume) < ACID_LEVEL_HANDBURN)
 		return NONE
-	if(user.gloves?.resistance_flags & (UNACIDABLE|ACID_PROOF))
+	if(user.gloves?.resistance_flags & (UNACIDABLE | ACID_PROOF))
 		return NONE
 
-	var/obj/item/bodypart/affecting = user.get_bodypart("[(user.active_hand_index % 2 == 0) ? "r" : "l" ]_arm")
-	if(!affecting?.receive_damage(0, 5))
-		return NONE
+	var/obj/item/bodypart/affecting = user.get_bodypart(!(user.active_hand_index % RIGHT_HANDS) ? BODY_ZONE_R_ARM : BODY_ZONE_L_ARM)
```

Comment:
this seems like an improper use of this define, the ! is also suspect

Reply:
Not using the define makes the code worse. I can avoid using !, but I find it to make the code slightly more confusing since most people use their right hand primarily.